tests/test_directory_resolver.py

```diff
@@ -0,0 +1,148 @@
+"""Test cases for DirectoryResolver."""
+#!/usr/bin/env python
+# coding=utf-8
+
+# Copyright New York University and the in-toto contributors
+# SPDX-License-Identifier: Apache-2.0
+
+
+import os
+import tempfile
+import unittest
+from pathlib import Path
+
+from in_toto.exceptions import PrefixError
+from in_toto.resolver import DirectoryResolver
+
+
+class TestDirectoryResolver(unittest.TestCase):
+    """Test directory resolver implementation."""
+
+    def test_regular_directory(self):
+        path = str(Path(__file__).parent / "resolver" / "dir_resolver")
+
+        resolver = DirectoryResolver(normalize_line_endings=True)
+        uri = f"dir:{path}"
+
+        # Expected hash calculated using:
+        # find . -type f | cut -c3- | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
+        expected_artifact_dict = {
+            uri: {
+                "sha256": "ecdbcdc6bd5d2966ad1f7595874fcd6f505abd3feaa97e27bae74bcb78c38e54"
+            }
+        }
+
+        artifact_dict = resolver.hash_artifacts([uri])
+        self.assertEqual(artifact_dict, expected_artifact_dict)
+
+    def test_regular_directory_with_lstrip(self):
+        parent_path = f"{Path(__file__).parent}/resolver/"
+        target_path = f"{parent_path}dir_resolver"
```

Comment:
This test (and the next one) won't work on a platform that requires backslashes as path separators. I think you should:

```suggestion
        lstrip_path = f"{Path(__file__).parent}/resolver/"
        target_path = str(Path(__file__).parent / "resolver" / "dir_resolver")
```

Reply:
... of course this requires normalising the path before lstripping.